pkgs/by-name/nu/nuget-to-json/nuget-to-json.sh

```diff
@@ -0,0 +1,119 @@
+#!@runtimeShell@
+# shellcheck shell=bash
+
+set -euo pipefail
+shopt -s nullglob
+
+export SSL_CERT_FILE=@cacert@/etc/ssl/certs/ca-bundle.crt
+export PATH="@binPath@:$PATH"
+# used for glob ordering of package names
+export LC_ALL=C
+
+if [ $# -eq 0 ]; then
+    >&2 echo "Usage: $0 <packages directory> [path to a file with a list of excluded packages] > deps.json"
+    exit 1
+fi
+
+pkgs=$1
+tmp=$(realpath "$(mktemp -td nuget-to-json.XXXXXX)")
+trap 'rm -r "$tmp"' EXIT
+
+excluded_list=$(realpath "${2:-/dev/null}")
+
+export DOTNET_NOLOGO=1
+export DOTNET_CLI_TELEMETRY_OPTOUT=1
+
+mapfile -t sources < <(dotnet nuget list source --format short | awk '/^E / { print $2 }')
+wait "$!"
+
+declare -a remote_sources
+declare -A base_addresses
+
+for index in "${sources[@]}"; do
+    if [[ -d "$index" ]]; then
+        continue
+    fi
+
+    remote_sources+=("$index")
+
+    base_address=$(
+        curl --compressed --netrc -fsL "$index" |
+            jq -r '.resources[] | select(."@type" == "PackageBaseAddress/3.0.0")."@id"'
+    )
+    if [[ ! "$base_address" == */ ]]; then
+        base_address="$base_address/"
+    fi
+    base_addresses[$index]="$base_address"
+done
+
+echo "["
+
+first=true
+cd "$pkgs"
+for package in *; do
+    [[ -d "$package" ]] || continue
+    cd "$package"
+    for version in *; do
+        id=$(xmlstarlet sel -t -v /_:package/_:metadata/_:id "$version"/*.nuspec)
+
+        if grep -qxF "$id.$version.nupkg" "$excluded_list"; then
+            continue
+        fi
+
+        # packages in the nix store should have an empty metadata file
+        used_source="$(jq -r 'if has("source") then .source else "" end' "$version"/.nupkg.metadata)"
```

Comment:
I think we can use jq's [alternation operator](https://jqlang.github.io/jq/manual/#alternative-operator) here? Similar to nix's `or`, python's `or`, JS's `||`, etc; it uses the alternate value if the lhs is missing, null, or false.

`jq -r '.source // ""'`

Reply:
Fixed.